File: frontend/lib/models/service.dart
/// Bike service model
class BikeService {
  final String id;
  final String name;
  final String? description;
  final double price;
  final int estimatedTime;
  final String estimatedTimeDisplay;
  final bool isActive;
  final DateTime createdAt;
  final DateTime updatedAt;

  BikeService({
    required this.id,
    required this.name,
    this.description,
    required this.price,
    required this.estimatedTime,
    required this.estimatedTimeDisplay,
    required this.isActive,
    required this.createdAt,
    required this.updatedAt,
  });

  factory BikeService.fromJson(Map<String, dynamic> json) {
    return BikeService(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: double.parse(json['price'].toString()),
      estimatedTime: json['estimated_time'],
      estimatedTimeDisplay: json['estimated_time_display'] ?? '${json['estimated_time']}m',
      isActive: json['is_active'] ?? true,
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'price': price,
      'estimated_time': estimatedTime,
      'is_active': isActive,
    };
  }
}

/// Service list response
class ServiceListResponse {
  final List<BikeService> services;
  final int total;

  ServiceListResponse({
    required this.services,
    required this.total,
  });

  factory ServiceListResponse.fromJson(Map<String, dynamic> json) {
    return ServiceListResponse(
      services: (json['services'] as List)
          .map((s) => BikeService.fromJson(s))
          .toList(),
      total: json['total'],
    );
  }
}

File: backend/app/schemas/service.py
"""
Service Schemas
Pydantic models for service-related request/response validation.
"""

from pydantic import BaseModel, Field, field_validator
from typing import Optional
from uuid import UUID
from datetime import datetime
from decimal import Decimal


class ServiceCreate(BaseModel):
    """Schema for creating a new service."""
    name: str = Field(..., min_length=2, max_length=100)
    description: Optional[str] = Field(None, max_length=500)
    price: Decimal = Field(..., ge=0)
    estimated_time: int = Field(..., gt=0, le=480)  # max 8 hours
    
    @field_validator('price')
    @classmethod
    def validate_price(cls, v):
        if v < 0:
            raise ValueError('Price must be positive')
        return round(v, 2)


class ServiceUpdate(BaseModel):
    """Schema for updating a service."""
    name: Optional[str] = Field(None, min_length=2, max_length=100)
    description: Optional[str] = Field(None, max_length=500)
    price: Optional[Decimal] = Field(None, ge=0)
    estimated_time: Optional[int] = Field(None, gt=0, le=480)
    is_active: Optional[bool] = None



class ServiceResponse(BaseModel):
    """Schema for service response."""
    id: UUID
    name: str
    description: Optional[str]
    price: Decimal
    estimated_time: int
    estimated_time_display: str
    is_active: bool
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class ServiceListResponse(BaseModel):
    """Schema for list of services response."""
    services: list[ServiceResponse]
    total: int
